src/lib/samples/zork-firefoxapp/index.js

```diff
@@ -1,16 +1,43 @@
 const {Cu} = require("chrome");
 var self = require("sdk/self");
 var {setTimeout} = require("sdk/timers");
+const {TextDecoder, OS} = Cu.import("resource://gre/modules/osfile.jsm", {});
 
 Cu.import(self.data.url("freedom-for-firefox/freedom-for-firefox.jsm"));
 
+const OPTIONS_FILE_PATH = '/zork-options';
+
+// Returns Promise<boolean>
+function checkIfMetricsEnabled() {
+  return OS.File.read(OPTIONS_FILE_PATH).then(
+    function onSuccess(array) {
```

Comment:
You don't need to name the function - this will save you a line, plus a level of indentation:
```ts
return OS.File.read(OPTIONS_FILE_PATH).then((array) => {
  // let decoder = , etc.
  // more code
});
```